src/general/world_effects/patch_events/GlobalGlaciationEvent.cs

```diff
@@ -213,6 +213,37 @@ private void AddIceChunks(Patch patch)
         }
     }
 
+    private int GetEventDuration()
+    {
+        switch (targetWorld.WorldSettings.ClimateInstability)
+        {
+            case WorldGenerationSettings.ClimateInstabilityEnum.Low:
+                return random.Next(Math.Max((int)(Constants.GLOBAL_GLACIATION_MIN_DURATION / 1.5f), 1),
+                    Math.Max((int)(Constants.GLOBAL_GLACIATION_MAX_DURATION / 1.5f), 1));
+            case WorldGenerationSettings.ClimateInstabilityEnum.High:
+                return random.Next((int)(Constants.GLOBAL_GLACIATION_MIN_DURATION * 1.5f),
+                    (int)(Constants.GLOBAL_GLACIATION_MAX_DURATION * 1.5f));
+            case WorldGenerationSettings.ClimateInstabilityEnum.Medium:
+            default:
+                return random.Next(Constants.GLOBAL_GLACIATION_MIN_DURATION,
+                    Constants.GLOBAL_GLACIATION_MAX_DURATION);
+        }
+    }
+
+    private float GetSunlightMultiplier()
```

Comment:
I think this code should have a warning that this may not change otherwise the ending the event blows up. I didn't read the world settings code change yet, but there should also be a warning there that ClimateInstability may not be changed due to the glaciation event requiring it to stay constant.